modules/library/src/main/scala/zio/elasticsearch/ElasticResult.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch
+
+import zio.prelude.{ForEachOps, ZValidation}
+import zio.schema.Schema
+import zio.{Task, ZIO}
+
+sealed trait ElasticResult[F[_]] {
+  def result[A: Schema]: Task[F[A]]
```

Comment:
What about using `.as[...]`?

Reply:
It makes sense, although we should consider that ZIO has an `as`, which means:
- we cannot make the ZIO extension methods `as` (if we decide to keep them at all)
- I wonder if it could be confusing having `as` with two different meanings in the same code-base

Reply:
What about `to[...]`?